pkg/gather/clusterconfig/image_registries.go

```diff
@@ -19,28 +22,51 @@ import (
 )
 
 // GatherClusterImageRegistry fetches the cluster Image Registry configuration
+// If the Image Registry configuration uses some PersistentVolumeClaim for the storage then the corresponding
+// PersistentVolume definition is gathered
 //
 // Location in archive: config/clusteroperator/imageregistry.operator.openshift.io/config/cluster.json
 // Id in config: image_registries
-func GatherClusterImageRegistry(g *Gatherer, c chan<- gatherResult){
+func GatherClusterImageRegistry(g *Gatherer, c chan<- gatherResult) {
 	defer close(c)
 	registryClient, err := imageregistryv1client.NewForConfig(g.gatherKubeConfig)
 	if err != nil {
 		c <- gatherResult{nil, []error{err}}
 		return
 	}
-	records, errors := gatherClusterImageRegistry(g.ctx, registryClient.ImageregistryV1())
+	gatherKubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		c <- gatherResult{nil, []error{err}}
+		return
+	}
+	records, errors := gatherClusterImageRegistry(g.ctx, registryClient.ImageregistryV1(), gatherKubeClient.CoreV1())
 	c <- gatherResult{records, errors}
 }
 
-func gatherClusterImageRegistry(ctx context.Context, registryClient imageregistryv1.ImageregistryV1Interface) ([]record.Record, []error) {
+func gatherClusterImageRegistry(ctx context.Context, registryClient imageregistryv1.ImageregistryV1Interface, coreClient corev1client.CoreV1Interface) ([]record.Record, []error) {
 	config, err := registryClient.Configs().Get(ctx, "cluster", metav1.GetOptions{})
 	if errors.IsNotFound(err) {
 		return nil, nil
 	}
 	if err != nil {
 		return nil, []error{err}
 	}
+	records := []record.Record{}
+	// if there is some PVC then try to gather used persistent volume
+	if config.Spec.Storage.PVC != nil {
+
+		pvcName := config.Spec.Storage.PVC.Claim
+		pv, err := findPVByPVCName(ctx, coreClient, pvcName)
+		if err != nil {
+			klog.Errorf("unable to find persistent volume: %s", err)
+		} else {
+			pvRecord := record.Record{
+				Name: fmt.Sprintf("config/persistentvolumes/%s", pv.Name),
```

Comment:
Wouldn't it make sense to have a subdir specifying where the PVs 'come from' ?
Something like: `Name: fmt.Sprintf("config/persistentvolumes/image-registry/%s", pv.Name),`

Reply:
Well I am not sure but I tend to say no. PV is not a namespaced resource AFAIK (guess it's cluster wide) and it's bound to specific PVC (this one is namespaced) that is then used in the IR config. AFAIK not the whole PV must be used when claimed by PVC, then it could be a bit misleading....

Reply:
Yep. PV are cluster-wide, not namespaced. Fine to have it as it is in the code.